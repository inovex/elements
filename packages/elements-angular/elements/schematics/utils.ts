import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
