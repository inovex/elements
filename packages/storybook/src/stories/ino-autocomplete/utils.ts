import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
