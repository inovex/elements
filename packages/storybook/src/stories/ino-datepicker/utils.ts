import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
