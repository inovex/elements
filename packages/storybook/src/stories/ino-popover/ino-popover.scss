.story-ino-popover {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 80px;

  ino-button {
    margin: 0 12px 12px 0;
  }

  ino-popover p {
    padding-left: 8px;
    padding-right: 8px;
  }

  .popover-content {
    padding: 10px;
    display: flex;
    flex-direction: row;
  }

  .styled-popover {
    padding: 24px;
    display: flex;
    justify-content: space-between;
    column-gap: 10px;
  }

  .interactive-popover {
    width: 300px;
    padding: 10px;
    display: flex;
    flex-direction: column;
    row-gap: 5px;
  }
}
