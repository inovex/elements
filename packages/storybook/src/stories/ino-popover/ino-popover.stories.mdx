import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-popover/readme.md";

<Meta title="Notification/ino-popover/Documentation" />

<Description markdown={readme} />
