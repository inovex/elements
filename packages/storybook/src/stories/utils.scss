/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
