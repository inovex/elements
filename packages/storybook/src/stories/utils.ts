import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
