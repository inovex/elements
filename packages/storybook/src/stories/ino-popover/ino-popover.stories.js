import { boolean, select } from '@storybook/addon-knobs';

import withStencilReadme from '_local-storybookcore/with-stencil-readme';

import componentReadme from '_local-elements/src/components/ino-popover/readme.md';
import './ino-popover.scss';

function subscribeToComponentEvents() {
  let popoverRef;

  const eventHandler = function (e) {
    console.log('EVENT: ', e);
    if (e.target) {
      const checkState = e.detail;

      e.target.setAttribute('ino-show', checkState);
    }
  };

  document.addEventListener('visibilityChanged', eventHandler);

  return () => {
    document.removeEventListener('checkedChange', eventHandler);
  };
}

export default {
  title: 'Notification/<ino-popover>',
  decorators: [
    withStencilReadme(componentReadme),
    (story) => {
      subscribeToComponentEvents();
      return story();
    },
  ],
};

export const DefaultUsage = () => /*html*/ `
    <div class="story-popover">

      <ino-button id="popover-target">Popover</ino-button>
      <ino-popover
        ino-show="false"
        ino-for="popover-target"
        ino-interactive="${boolean('ino-interactive', false)}"
        ino-placement="${select(
          'ino-placement',
          ['top', 'right', 'bottom', 'left'],
          'top'
        )}"
        ino-color-scheme="${select(
          'ino-color-scheme',
          ['primary', 'secondary', 'light', 'transparent'],
          'transparent'
        )}"
        ino-trigger="${select(
          'ino-trigger',
          [
            'mouseenter',
            'focus',
            'click',
            'mouseenter focus',
            'mouseenter click',
            'focus click',
            'mouseenter focus click',
          ],
          'mouseenter focus'
        )}">
        <div class="styled-popover"><ino-icon ino-icon="user"></ino-icon> This is a styled popover.</div>
      </ino-popover>

      <h4>Placements</h4>
      <ino-popover ino-placement="left" ino-for="popover-positions-target">This is a simple popover on the left</ino-popover>
      <ino-popover ino-placement="right" ino-for="popover-positions-target">This is a simple popover on the right</ino-popover>
      <ino-popover ino-placement="top" ino-for="popover-positions-target">This is a simple popover on the top</ino-popover>
      <ino-popover ino-placement="bottom" ino-for="popover-positions-target">This is a simple popover on the bottom</ino-popover>
      <ino-button class="placement-button" id="popover-positions-target">Popover</ino-button>

      <h4>Triggers</h4>
      <div class="row">
      <ino-button id="popover-hover">Mouseenter</ino-button>
      <ino-popover ino-placement="left" ino-for="popover-hover" ino-trigger="mouseenter">This popover occurs on mouseenter</ino-popover>

      <ino-button id="popover-focus">Focus</ino-button>
      <ino-popover ino-placement="top" ino-for="popover-focus" ino-trigger="focus">This popover occurs on focus.</ino-popover>

      <ino-button id="popover-click">Click</ino-button>
      <ino-popover ino-placement="right" ino-for="popover-click" ino-trigger="click">This popover occurs on click.</ino-popover>
</div>
      <h4>Interactions</h4>
            <div class="row">
      <ino-button id="popover-non-interactive">Non-Interactive content</ino-button>
      <ino-popover ino-interactive="false" ino-for="popover-non-interactive" ino-trigger="click" ino-color-scheme="transparent">
        <div id="interactive-demo-container">
            <p>I'm not interactive. I will close on click. I should only be used for non-interactive content like this text.</p>
        </div>
      </ino-popover>

      <ino-button id="popover-interactive">Interactive content</ino-button>
      <ino-popover ino-interactive="true" ino-for="popover-interactive" ino-trigger="click" ino-color-scheme="transparent">
        <div id="interactive-demo-container">
            <ino-button>You can click me without closing this popover!</ino-button>
        </div>
      </ino-popover>
      </div>
      <h4>Controlled Popover</h4>
      <ino-checkbox id="controlled-checkbox" checked="true">Uncheck to hide / check to show</ino-checkbox>
      <ino-popover id="controlled-popover" ino-placement="bottom" ino-for="controlled-checkbox" ino-show="true">
        I'm a controlled popover
      </ino-popover>
    </div>
  `;
