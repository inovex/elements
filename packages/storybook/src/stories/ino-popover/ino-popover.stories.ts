import { useEffect } from '@storybook/client-api';
import { html } from 'lit-html';
import { defaultDecorator, withColorScheme, withIconControl } from '../utils';
import './ino-popover.scss';

export default {
  title: `Notification/ino-popover`,
  component: 'ino-popover',
  parameters: {
    actions: {
      handles: ['visibleChange ino-popover'],
    },
  },
  decorators: [(story) => defaultDecorator(story, 'story-ino-popover')],
};

export const Playground = (args) => html`
  <ino-button id="popover-target">Popover</ino-button>
  <ino-popover
    color-scheme="${args.colorScheme}"
    controlled="${args.controlled}"
    distance="${args.distance}"
    for="${args.for}"
    interactive="${args.interactive}"
    placement="${args.placement}"
    trigger="${args.trigger}"
    visible="${args.visible}"
  >
    <div class="styled-popover">
      <ino-icon icon="user"></ino-icon> This is a styled popover.
    </div>
  </ino-popover>
`;
Playground.args = {
  controlled: false,
  distance: 10,
  for: 'popover-target',
  interactive: false,
  placement: 'top',
  trigger: 'mouseenter focus',
  colorScheme: 'transparent',
  visible: false,
};
Playground.argTypes = {
  placement: {
    control: {
      type: 'select',
    },
    options: ['top', 'right', 'bottom', 'left'],
  },
  trigger: {
    control: {
      type: 'select',
    },
    options: [
      'mouseenter',
      'focus',
      'click',
      'mouseenter focus',
      'mouseenter click',
      'focus click',
      'mouseenter focus click',
    ],
  },
  colorScheme: {
    control: {
      type: 'select',
    },
    options: ['primary', 'secondary', 'light', 'transparent'],
  },
};

export const Placement = () => html`
  <ino-popover placement="left" for="popover-positions-target">
    <div class="popover-content">This is a simple popover on the left.</div>
  </ino-popover>
  <ino-popover placement="right" for="popover-positions-target">
    <div class="popover-content">This is a simple popover on the right.</div>
  </ino-popover>
  <ino-popover placement="top" for="popover-positions-target">
    <div class="popover-content">This is a simple popover on the top.</div>
  </ino-popover>
  <ino-popover placement="bottom" for="popover-positions-target">
    <div class="popover-content">This is a simple popover on the bottom.</div>
  </ino-popover>
  <ino-button class="placement-button" id="popover-positions-target"
    >Popover</ino-button
  >
`;

export const Interactions = () => html`
  <ino-button id="popover-non-interactive">Non-Interactive content</ino-button>
  <ino-popover
    for="popover-non-interactive"
    trigger="click"
    color-scheme="transparent"
  >
    <div class="popover-content">
      I'm not interactive. I will close on click. I should only be used for
      non-interactive content like this text.
    </div>
  </ino-popover>
  <ino-popover
    trigger="click"
    placement="bottom"
    for="popover-interactive-target"
    color-scheme="transparent"
    interactive
  >
    <div class="interactive-popover">
      <p>I'm interactive. You can click me without closing this popover!</p>
      <ino-button>Button not closing the popover</ino-button>
    </div>
  </ino-popover>
  <ino-button class="placement-button" id="popover-interactive-target"
    >Interactive Content</ino-button
  >
`;

export const Trigger = () => html`
  <ino-button id="popover-hover-focus">Hover & focus</ino-button>
  <ino-popover for="popover-hover-focus" placement="top">
    <div class="popover-content">This popover occurs on hover and focus.</div>
  </ino-popover>

  <ino-button id="popover-click">Click</ino-button>
  <ino-popover for="popover-click" trigger="click" placement="bottom">
    <div class="popover-content">This popover occurs on click.</div>
  </ino-popover>

  <ino-button id="popover-focus">Focus</ino-button>
  <ino-popover
    class="popover-content"
    placement="right"
    for="popover-focus"
    trigger="focus"
  >
    <div class="popover-content">This popover occurs on focus.</div>
  </ino-popover>
`;

export const ControlledPopover = () => {
  useEffect(() => {
    const eventHandler = (e) => {
      e.target?.setAttribute('visible', e.detail);
      (document.querySelector(
        '#controlled-checkbox'
      ) as HTMLInputElement).checked = e.detail;
    };

    document.addEventListener('visibleChanged', eventHandler);
    return () => document.removeEventListener('visibleChanged', eventHandler);
  });

  return html`
    <ino-popover
      id="controlled-popover"
      placement="left"
      controlled
      trigger="click"
    >
      <ino-checkbox id="controlled-checkbox" slot="popover-trigger"
        >Uncheck to hide / check to show</ino-checkbox
      >
      <div class="popover-content">I'm a controlled popover</div>
    </ino-popover>
  `;
};
