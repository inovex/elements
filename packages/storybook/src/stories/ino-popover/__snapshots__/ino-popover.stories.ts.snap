// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification/ino-popover AttachToBody smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-popover">
    <ino-button id="popover-3"
                class="hydrated"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Popover
        </span>
      </button>
    </ino-button>
    <ino-popover color-scheme="transparent"
                 controlled="false"
                 distance="10"
                 for="popover-3"
                 interactive="false"
                 followcursor="undefined"
                 placement="top"
                 trigger="mouseenter focus"
                 visible="false"
                 hide-on-blur="false"
                 hide-on-esc="false"
                 class="hydrated"
    >
      <div>
      </div>
    </ino-popover>
  </div>
</div>
`;

exports[`Notification/ino-popover ColorScheme smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-popover">
    <ino-button id="popover-5"
                class="hydrated"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Popover
        </span>
      </button>
    </ino-button>
    <ino-popover color-scheme="secondary"
                 controlled="false"
                 distance="10"
                 for="popover-5"
                 interactive="false"
                 followcursor="undefined"
                 placement="top"
                 trigger="mouseenter focus"
                 visible="false"
                 hide-on-blur="false"
                 hide-on-esc="false"
                 class="hydrated"
    >
      <div>
      </div>
    </ino-popover>
  </div>
</div>
`;

exports[`Notification/ino-popover Distance smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-popover">
    <ino-button id="popover-4"
                class="hydrated"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Popover
        </span>
      </button>
    </ino-button>
    <ino-popover color-scheme="transparent"
                 controlled="false"
                 distance="30"
                 for="popover-4"
                 interactive="false"
                 followcursor="undefined"
                 placement="top"
                 trigger="mouseenter focus"
                 visible="false"
                 hide-on-blur="false"
                 hide-on-esc="false"
                 class="hydrated"
    >
      <div>
      </div>
    </ino-popover>
  </div>
</div>
`;

exports[`Notification/ino-popover FollowCursor smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-popover">
    <ino-button id="popover-7"
                class="hydrated"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Popover
        </span>
      </button>
    </ino-button>
    <ino-popover color-scheme="transparent"
                 controlled="false"
                 distance="10"
                 for="popover-7"
                 interactive="false"
                 followcursor="horizontal"
                 placement="top"
                 trigger="mouseenter focus"
                 visible="false"
                 hide-on-blur="false"
                 hide-on-esc="false"
                 class="hydrated"
    >
      <div>
      </div>
    </ino-popover>
  </div>
</div>
`;

exports[`Notification/ino-popover Interactions smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-popover">
    <ino-popover trigger="click"
                 for="popover-interactive-target"
                 color-scheme="transparent"
                 interactive
                 placement="left"
                 class="hydrated"
    >
      <div>
      </div>
    </ino-popover>
    <ino-button class="placement-button hydrated"
                id="popover-interactive-target"
                aria-expanded="false"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Interactive Content
        </span>
      </button>
    </ino-button>
  </div>
</div>
`;

exports[`Notification/ino-popover Placement smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-popover">
    <ino-button id="popover-2"
                class="hydrated"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Popover
        </span>
      </button>
    </ino-button>
    <ino-popover color-scheme="transparent"
                 controlled="false"
                 distance="10"
                 for="popover-2"
                 interactive="false"
                 followcursor="undefined"
                 placement="right"
                 trigger="mouseenter focus"
                 visible="false"
                 hide-on-blur="false"
                 hide-on-esc="false"
                 class="hydrated"
    >
      <div>
      </div>
    </ino-popover>
  </div>
</div>
`;

exports[`Notification/ino-popover Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-popover">
    <ino-button id="popover-1"
                class="hydrated"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Popover
        </span>
      </button>
    </ino-button>
    <ino-popover color-scheme="transparent"
                 controlled="false"
                 distance="10"
                 for="popover-1"
                 interactive="false"
                 followcursor="undefined"
                 placement="top"
                 trigger="mouseenter focus"
                 visible="false"
                 hide-on-blur="false"
                 hide-on-esc="false"
                 class="hydrated"
    >
      <div>
      </div>
    </ino-popover>
  </div>
</div>
`;

exports[`Notification/ino-popover Trigger smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-popover">
    <ino-button id="popover-6"
                class="hydrated"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Popover
        </span>
      </button>
    </ino-button>
    <ino-popover color-scheme="transparent"
                 controlled="false"
                 distance="10"
                 for="popover-6"
                 interactive="false"
                 followcursor="undefined"
                 placement="top"
                 trigger="click"
                 visible="false"
                 hide-on-blur="false"
                 hide-on-esc="false"
                 class="hydrated"
    >
      <div>
      </div>
    </ino-popover>
  </div>
</div>
`;

exports[`Notification/ino-popover Visible smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-popover">
    <ino-popover id="controlled-popover"
                 placement="left"
                 controlled="true"
                 trigger="click"
                 visible="false"
    >
      <ino-checkbox id="controlled-checkbox"
                    slot="popover-trigger"
      >
        Uncheck to hide / check to show
      </ino-checkbox>
      <div>
        <div class="ino-popover ino-popover--color-scheme-primary ino-popover__content"
             role="tooltip"
        >
          <div class="popover-content">
            I'm a controlled popover
          </div>
        </div>
      </div>
    </ino-popover>
  </div>
</div>
`;
